src/e2e/puppeteer/__tests__/toggleSidebar.ts

```diff
@@ -0,0 +1,29 @@
+import click from '../helpers/click'
+import getSelection from '../helpers/getSelection'
+import paste from '../helpers/paste'
+import waitForEditable from '../helpers/waitForEditable'
+
+vi.setConfig({ testTimeout: 20000, hookTimeout: 20000 })
+
+describe('all platforms', () => {
+  it('on toggleSidebar, the caret offset should be preserved once it closes.', async () => {
```

Comment:
It might be worthwhile to write 3 tests for this, covering the 3 different ways to close the sidebar:
* Hit Escape
* Click hamburger menu
* Click on empty space in the editor